src/classes/AFFL_AccRecordType_TDTM.cls

```diff
@@ -62,36 +71,47 @@ public class AFFL_AccRecordType_TDTM extends TDTM_Runnable {
+    	 	
+             for (Integer i = 0; i < newlist.size(); i++) {
     	 	    SObject so = newlist[i];
     	 	    //AFTER UPDATE
     	 		if(triggerAction == TDTM_Runnable.Action.AfterUpdate) {
 	    	 		Account acc = (Account)so;
 			        Account oldAcc = (Account)oldList[i];
 
-			        if(acc.RecordTypeId != oldAcc.RecordTypeId) {
-			        	AFFL_MultiRecordTypeMapper afflMapper = new AFFL_MultiRecordTypeMapper();
+                    // If the record type isn't changing, move on
+                    if(acc.RecordTypeId == oldAcc.RecordTypeId) {
+                        continue;
+                    }
+
+			        AFFL_MultiRecordTypeMapper afflMapper = new AFFL_MultiRecordTypeMapper();
 
-			        	//Find the lookup field that matches the new record type
-                        String keyAfflFieldNewType = afflMapper.getKeyAfflFieldByAccRecordType(acc.RecordTypeId);
+                    //Find the lookup field that matches the new record type
+                    String keyAfflFieldNewType = afflMapper.getKeyAfflFieldByAccRecordType(acc.RecordTypeId);
 
-			        	//Find the lookup field that matches the old record type
-				        String keyAfflFieldOldType = afflMapper.getKeyAfflFieldByAccRecordType(oldAcc.RecordTypeId);
+                    //Find the lookup field that matches the old record type
+                    String keyAfflFieldOldType = afflMapper.getKeyAfflFieldByAccRecordType(oldAcc.RecordTypeId);
 
-				       //Checking if keyAfflFieldOldType is null instead of acc.RecordTypeId because apparently that field
-				       //has a value even if the Account is created without populating that field in a test.
-				       if(keyAfflFieldOldType != null) {
-				        	//Add the account info to the list that tracks accounts that have changed type
-				        	accsTypeChanged.add(new AccNewOldType(acc.ID, keyAfflFieldOldType, keyAfflFieldNewType));
-			        	} else if (keyAfflFieldNewType != null) {
-			        	    accsTypeAdded.put(acc.Id, keyAfflFieldNewType);
-			        	}
-			        }
+                    //Checking if keyAfflFieldOldType is null instead of acc.RecordTypeId because apparently that field
+                    //has a value even if the Account is created without populating that field in a test.
+                    if(keyAfflFieldOldType != null) {
+                        //Add the account info to the list that tracks accounts that have changed type
+                        accsTypeChanged.add(new AccNewOldType(acc.ID, keyAfflFieldOldType, keyAfflFieldNewType));
+                    } else if (keyAfflFieldNewType != null) {
+                        accsTypeAdded.put(acc.Id, keyAfflFieldNewType);
+                    }
     	 		}
 		    }
+
 		    updatePrimaryAfflsOnChange(accsTypeChanged, dmlWrapper);
             updatePrimaryAfflsOnAdd(accsTypeAdded, dmlWrapper);
+
+            TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.AFFL_AccRecordType_TDTM, true);
```

Comment:
This "feels" like a significant change (I know you warned me about it). We would want to get this tested, and any products built on top of EDA, asap. Would you mind writing up something that discusses this implementation so that we can give other's a heads up?

Reply:
I'll have to think about this more. I did write up a Quip doc for bulk data loader testing. I think the biggest change is that instead of grouping the affiliation record updates at the end with the other potential record updates, we're doing it here.  The likelihood that account record types are being changed in bulk at the same time as other updates (like address updates) is probably pretty low.

I expect that if anyone is using the bulk api to perform up to 10,000 account record type updates today, they are probably already running into potential CPU limits and DML limits. But maybe we need to call out something around this in the release notes?

Reply:
Interesting - we call out some best practices in this article with regards to TDTM and bulk data processing / API users already: https://powerofus.force.com/s/article/HEDA-Disable-Trigger-Handlers

Reply:
@robertbwatson that's a good find. I say we go ahead as planned. I tend to agree that flipping the account RTs in bulk along with other triggers is probably an edge case. I think we will be okay. I know you're working on other items like this at the moment as well, do we plan on doing the same sort of thing there?

Reply:
@lifewithryan Yeah, I plan on writing up some sort of document for bulk data loader testing for all of the other WIs as well.

Reply:
@lifewithryan I have been using the Data Import Wizard rather than the DataLoader tool for testing. Pls let me know if that's OK.

Reply:
I'm getting this error with 2000 records : 

> CANNOT_INSERT_UPDATE_ACTIVATE_ENTITY:hed.TDTM_Account: System.LimitException: Apex CPU time limit exceeded:--

Do I need to disable a particular trigger handler, as referred to in the doc linked above?

Reply:
@dgamboni I'd be curious to see if you get the error with Dataloader as well. Import Wizard is a bit different IIRC.

Reply:
Not surprising that a CPU timeout is being thrown when working with a large number of records, but I do want to investigate further to make sure that we get about the same results when working with the master branch.

This could spawn a few entirely different work items that tries to improve the efficiency of our code - the fact that only X number of accounts can be inserted/updated at one time without timing out is a bit of a concern to me. Many organizations probably have multiple other packages that also perform logic on account insert/updates so the failure rate would be even higher in a customer org. We want our code to be as fast as possible, so it might be worth spending some time to determine where we might be able to make our code better.